distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+    }
+
+    @Test
+    public void testSlowResponse() throws Exception {
+        RestitoRetrievalResponse slowAction = new RestitoRetrievalResponse("testData", 2000);
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), slowAction.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+    }
+
+    // Tests that if the endpoint disconnects twice, the retrieval retries both times
+    @Test
+    public void testRetrievalReliablility() throws Exception {
+        RestitoRetrievalResponse slowFail = new RestitoRetrievalResponse("testData", 500, 2);
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), slowFail.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+    }
+
+    // Tests that is the endpoint disconnects 3 times, the retrieval fails after 3 attempts
+    @Test
+    public void testRetreivalReliabilityFails() throws Exception {
+        RestitoRetrievalResponse slowFail = new RestitoRetrievalResponse("testData", 500, 3);
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), slowFail.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that product retrieval fails from the csw stub server.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().statusCode(500).contentType("text/plain")
+                .body(containsString("cannot retrieve product"));
+        // @formatter:on
```

Comment:
What should I be verifying here? This is only testing that the ReliableResourceDownloader only retries 3 times before failing - as proven by passing the previous test which has 2 planned disconnects that this test which has 3.